### PR TITLE
[ML] move find file structure finder in Rest high Level client to its new endpoint and plugin

### PR DESCRIPTION
**Pull request by benwtrent**
Find file structure finder is now its own plugin, and separated from the ml plugin.

This commit updates the rest high level client to reflect this.

Additionally, this adjusts the internal and client object names from `FileStructure` to the more general `TextStructure`

### CODE REVIEW

File: client/rest-high-level/src/main/java/org/elasticsearch/client/RestHighLevelClient.java [hunk 1]
```diff
@@ -454,6 +455,16 @@ public AsyncSearchClient asyncSearch() {
         return asyncSearchClient;
     }
 
+    /**
+     * A wrapper for the {@link RestHighLevelClient} that provides methods for accessing the Elastic Text Structure APIs.
+     * <p>
+     * See the <a href="https://www.elastic.co/guide/en/elasticsearch/reference/current/text-structure.html"> X-Pack APIs on elastic.co</a>
```
Comment (benwtrent) [inline client/rest-high-level/src/main/java/org/elasticsearch/client/RestHighLevelClient.java:461]:
```suggestion
     * See the <a href="https://www.elastic.co/guide/en/elasticsearch/reference/current/find-structure.html"> X-Pack APIs on elastic.co</a>
```

File: client/rest-high-level/src/main/java/org/elasticsearch/client/TextStructureClient.java [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+import org.elasticsearch.action.ActionListener;
+import org.elasticsearch.client.textstructure.FindStructureRequest;
+import org.elasticsearch.client.textstructure.FindStructureResponse;
+
+
+/**
+ * Text Structure API client wrapper for the {@link RestHighLevelClient}
+ * <p>
+ * See the <a href="https://www.elastic.co/guide/en/elasticsearch/reference/current/ml-apis.html">
+ * X-Pack Text Structure APIs </a> for additional information.
```
Comment (benwtrent) [inline client/rest-high-level/src/main/java/org/elasticsearch/client/TextStructureClient.java:33]:
```suggestion
 * See the <a href="https://www.elastic.co/guide/en/elasticsearch/reference/current/find-structure.html">
 * X-Pack Text Structure APIs </a> for additional information.
```

File: client/rest-high-level/src/main/java/org/elasticsearch/client/TextStructureClient.java [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+
+    TextStructureClient(RestHighLevelClient restHighLevelClient) {
+        this.restHighLevelClient = restHighLevelClient;
+    }
+
+    /**
+     * Finds the structure of a file
+     * <p>
+     * For additional info
+     * see <a href="https://www.elastic.co/guide/en/elasticsearch/reference/current/ml-find-file-structure.html">
```
Comment (benwtrent) [inline client/rest-high-level/src/main/java/org/elasticsearch/client/TextStructureClient.java:47]:
```suggestion
     * see <a href="https://www.elastic.co/guide/en/elasticsearch/reference/current/find-structure.html">
```

File: client/rest-high-level/src/main/java/org/elasticsearch/client/TextStructureClient.java [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+            options,
+            FindStructureResponse::fromXContent,
+            Collections.emptySet());
+    }
+
+    /**
+     * Finds the structure of a file asynchronously and notifies the listener on completion
+     * <p>
+     * For additional info
+     * see <a href="https://www.elastic.co/guide/en/elasticsearch/reference/current/ml-find-file-structure.html">
```
Comment (benwtrent) [inline client/rest-high-level/src/main/java/org/elasticsearch/client/TextStructureClient.java:67]:
```suggestion
     * see <a href="https://www.elastic.co/guide/en/elasticsearch/reference/current/find-structure.html">
```

File: docs/java-rest/high-level/textstructure/find-structure.asciidoc [hunk 5]
```diff
@@ -1,30 +1,30 @@
...
+Determines the structure of text and other information that will be
 useful to import its contents to an {es} index. It accepts a +{request}+ object
 and responds with a +{response}+ object.
 
 [id="{upid}-{api}-request"]
-==== Find file structure request
+==== Find structure request
 
-A sample from the beginning of the file (or the entire file contents if
+A sample from the beginning of the text (or the entire text contents if
```
Comment (davidkyle) [inline docs/java-rest/high-level/textstructure/find-structure.asciidoc:19]: This makes sentence makes less sense once file is replaced with text especially the entire contents bit. 

```suggestion
A sample of the the text to analyze must be added to the +{request}+ object using the....
```

File: client/rest-high-level/src/main/java/org/elasticsearch/client/TextStructureClient.java [hunk 6]
```diff
@@ -0,0 +1,85 @@
...
+    TextStructureClient(RestHighLevelClient restHighLevelClient) {
+        this.restHighLevelClient = restHighLevelClient;
+    }
+
+    /**
+     * Finds the structure of a file
+     * <p>
+     * For additional info
+     * see <a href="https://www.elastic.co/guide/en/elasticsearch/reference/current/find-structure.html">
+     *     Text Structure Find Structure documentation</a>
```
[the remark this answers is not in the document]
Comment (benwtrent) [reply client/rest-high-level/src/main/java/org/elasticsearch/client/TextStructureClient.java:48]: yeah!

File: client/rest-high-level/src/main/java/org/elasticsearch/client/textstructure/FindStructureRequest.java [hunk 7]
```diff
@@ -7,7 +7,7 @@
  * not use this file except in compliance with the License.
  * You may obtain a copy of the License at
  *
- *    http://www.apache.org/licenses/LICENSE-2.0
+ *     http://www.apache.org/licenses/LICENSE-2.0
```
[the remark this answers is not in the document]
Comment (benwtrent) [reply client/rest-high-level/src/main/java/org/elasticsearch/client/textstructure/FindStructureRequest.java:10]: auto formatting made this change? I can revert to see...

File: client/rest-high-level/src/main/java/org/elasticsearch/client/textstructure/structurefinder/FieldStats.java [hunk 8]
```diff
@@ -7,7 +7,7 @@
  * not use this file except in compliance with the License.
  * You may obtain a copy of the License at
  *
- *    http://www.apache.org/licenses/LICENSE-2.0
+ *     http://www.apache.org/licenses/LICENSE-2.0
```
Comment (benwtrent) [inline client/rest-high-level/src/main/java/org/elasticsearch/client/textstructure/structurefinder/FieldStats.java:10]:
```suggestion
 *    http://www.apache.org/licenses/LICENSE-2.0
```

File: client/rest-high-level/src/main/java/org/elasticsearch/client/textstructure/structurefinder/TextStructure.java [hunk 9]
```diff
@@ -7,7 +7,7 @@
  * not use this file except in compliance with the License.
  * You may obtain a copy of the License at
  *
- *    http://www.apache.org/licenses/LICENSE-2.0
+ *     http://www.apache.org/licenses/LICENSE-2.0
```
Comment (benwtrent) [inline client/rest-high-level/src/main/java/org/elasticsearch/client/textstructure/structurefinder/TextStructure.java:10]:
```suggestion
 *    http://www.apache.org/licenses/LICENSE-2.0
```

File: client/rest-high-level/src/test/java/org/elasticsearch/client/textstructure/FindStructureResponseTests.java [hunk 10]
```diff
@@ -7,7 +7,7 @@
  * not use this file except in compliance with the License.
  * You may obtain a copy of the License at
  *
- *    http://www.apache.org/licenses/LICENSE-2.0
+ *     http://www.apache.org/licenses/LICENSE-2.0
```
Comment (benwtrent) [inline client/rest-high-level/src/test/java/org/elasticsearch/client/textstructure/FindStructureResponseTests.java:10]:
```suggestion
 *    http://www.apache.org/licenses/LICENSE-2.0
```

File: client/rest-high-level/src/test/java/org/elasticsearch/client/textstructure/structurefinder/FieldStatsTests.java [hunk 11]
```diff
@@ -7,7 +7,7 @@
  * not use this file except in compliance with the License.
  * You may obtain a copy of the License at
  *
- *    http://www.apache.org/licenses/LICENSE-2.0
+ *     http://www.apache.org/licenses/LICENSE-2.0
```
Comment (benwtrent) [inline client/rest-high-level/src/test/java/org/elasticsearch/client/textstructure/structurefinder/FieldStatsTests.java:10]:
```suggestion
 *    http://www.apache.org/licenses/LICENSE-2.0
```

File: client/rest-high-level/src/test/java/org/elasticsearch/client/textstructure/structurefinder/TextStructureTests.java [hunk 12]
```diff
@@ -7,7 +7,7 @@
  * not use this file except in compliance with the License.
  * You may obtain a copy of the License at
  *
- *    http://www.apache.org/licenses/LICENSE-2.0
+ *     http://www.apache.org/licenses/LICENSE-2.0
```
Comment (benwtrent) [inline client/rest-high-level/src/test/java/org/elasticsearch/client/textstructure/structurefinder/TextStructureTests.java:10]:
```suggestion
 *    http://www.apache.org/licenses/LICENSE-2.0
```

File: client/rest-high-level/src/main/java/org/elasticsearch/client/textstructure/FindStructureResponse.java [hunk 13]
```diff
@@ -7,7 +7,7 @@
  * not use this file except in compliance with the License.
  * You may obtain a copy of the License at
  *
- *    http://www.apache.org/licenses/LICENSE-2.0
+ *     http://www.apache.org/licenses/LICENSE-2.0
```
Comment (davidkyle) [inline client/rest-high-level/src/main/java/org/elasticsearch/client/textstructure/FindStructureResponse.java:10]:
```suggestion
 *    http://www.apache.org/licenses/LICENSE-2.0
```

**Review by davidkyle — APPROVED** [review]
LGTM

**Comment by benwtrent** [comment]
run elasticsearch-ci/bwc

**Comment by benwtrent** [comment]
@elasticmachine update branch

**Comment by benwtrent** [comment]
@elasticmachine update branch

**Pull request merged**